File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        if (target<nums[l]) return 0;
        if (target>nums[r]) return nums.length;
        while (l<r) {
            int val = l+(r-l)/2;
            if (nums[val] == target) return val;
            if (nums[val] < target) l = val+1;
            else r = val;
        }    
        return l;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n=nums.size();
        int low=0;
        int high=n-1;
        int x=0;
        int ans=n;
        while(low<=high){
            int mid=(low+high)/2;
            if(nums[mid]==target){
                return mid;                         
            }
            else if(nums[mid]>target){
                ans=mid;
                high=mid-1;
            }
            else if(nums[mid]<target){
                low=mid+1;
            }
            
            
            
        }
        if(target>nums[n-1]){
            return n;
        }
        if(target<nums[0]){
            return 0;
        }
        return ans;
        
    }
};